### PR TITLE
chore: Add Issue Forms

### PR DESCRIPTION
**Pull request by sumeshir26**
### Description
Issue forms added

### Resolved / Related Issues
_None_

### Type of Change

- [ ] Feature
- [ ] Bug Fix
- [x] Documentation
- [ ] Localization

### Screenshots (optional)
Add screenshots here.

### CODE REVIEW

**Comment by im-coder-lg** [comment]
Heyyyyy... Wait a minute... In the YML files, can you add an assignees placeholder and add @Melvin-Abraham to it? It was possible on the Markdown edition so it might be possible here too, right? Just below the `labels:` tag, you can add it if it really does exist. Then, it doesn't need extra workflows to assign to a person.

**Comment by im-coder-lg** [comment]
Figured some things out with issue forms. 

1. Multiple assignees
2. Drop-down options (supports multiple options too)
3. Checkboxes

And I found a method to prevent blank issues. If you get the `.github/ISSUE_TEMPLATES/config.yml` and enter this:
```yaml
blank_issues_enabled: false
```
It could prevent blank issues. YOu can put dropdown for versions of GAUDC, operating systems, their versions and build numbers, and et cetra.

**Comment by im-coder-lg** [comment]
Maybe I can help here. Do you need help in anything related here?

**Comment by sumeshir26** [comment]
@im-coder-lg Please help in the assignees part(I am very basic in yml files)

**Comment by im-coder-lg** [comment]
Okay, startin' the job now!

**Comment by im-coder-lg** [comment]
@sumeshir26 **`danger levels`** see this:
image
We got a situation in that. Happened to me also when I was testing it.

**Comment by im-coder-lg** [comment]
Let's try starting the form with markdown. I will open a PR in *your* fork, not here.

**Comment by im-coder-lg** [comment]
Maybe all it needed was double quotes. I hit that `Learn More` button and got some solutions(?), and I am testing. Merge that when I say.

**Comment by im-coder-lg** [comment]
@sumeshir26 all the forms we collaborated on are useless. The original perfect version:
```yaml
name: Bug Report
description: File a bug report
title: "[Bug]: "
labels: [bug, triage]
assignees:
- monalisa
- nat
body:
- type: markdown
  attributes:
    value: |
      Thanks for taking the time to fill out this bug report!
- type: input
  id: contact
  attributes:
    label: Contact Details
    description: How can we get in touch with you if we need more info?
    placeholder: ex. email@example.com
  validations:
    required: false
- type: textarea
  id: what-happened
  attributes:
    label: What happened?
    description: Also tell us, what did you expect to happen?
    placeholder: Tell us what you see!
    value: "A bug happened!"
  validations:
    required: true
- type: dropdown
  id: version
  attributes:
    label: Version
    description: What version of our software are you running?
    options:
    - 1.0.2 (Default)
    - 1.0.3 (Edge)
  validations:
    required: true
- type: dropdown
  id: browsers
  attributes:
    label: What browsers are you seeing the problem on?
    multiple: true
    options:
    - Firefox
    - Chrome
    - Safari
    - Microsoft Edge
- type: textarea
  id: logs
  attributes:
    label: Relevant log output
    description: Please copy and paste any relevant log output. This will be automatically formatted into code, so no need for backticks.
    render: shell
- type: checkboxes
  id: terms
  attributes:
    label: Code of Conduct
    description: By submitting this issue, you agree to follow our [Code of Conduct](https://example.com)
    options:
    - label: I agree to follow this project's Code of Conduct
      required: true
```
This is an example but it is perfect, I was testing it. See the screenshots below:
image
image

This means that both of us did a LOT of YAML mistakes! I am going to use this code as base and make the PERFECT bug report form.

**Comment by sumeshir26** [comment]
@im-coder-lg So do you want me to merge your PR?

**Comment by im-coder-lg** [comment]
No. I am still doing that work, and it is not done(see the errors in bug-report form), so I spinned up a Gitpod environment and it recognised that the YAML file is a GitHub Issue form. So, it is pinpointing the errors to the files. Link to the Gitpod environment:
https://beige-hornet-4nyidy13.ws-us15.gitpod.io
You can notice that there are so many errors. I am fixing that, so wait for the go-ahead. When Gitpod clears all the errors, you can merge it here.

**Comment by im-coder-lg** [comment]
Seems like it is fixed, starting the work on the feature request form

**Comment by im-coder-lg** [comment]
Bug report form is finally fixed @sumeshir26 wait for `hour or two and I will fix the feature request form.

**Comment by im-coder-lg** [comment]
I don't think I need an hour, give 15 minutes and I will make everything ready.

**Comment by im-coder-lg** [comment]
Yup, @sumeshir26 you can merge the form PR. All is done :)

**Comment by im-coder-lg** [comment]
Now, only @Melvin-Abraham has to see this and merge it. If you want to prevent blank issues, you can add a `./.github/ISSUE_TEMPLATES/config.yml` file that configures the templates and add this line:
```yaml
blanl_issues_enabled: false
```
That can stop blank issues. Then, if anybody needs

**Comment by sumeshir26** [comment]
@im-coder-lg Do you think we should add a note for common issues like 'invlid_grant' and 'Tried calling start before the ready() event!'?

**Comment by sumeshir26** [comment]
Also, I dont think we should disable blank issues since forms are very limited in how the user can enter data

**Comment by im-coder-lg** [comment]
That can be good, if there was a website for the project. I can use Jekyll or CODEDOC but I need a separate branch, a separate workflow and all that things. Point is, it's a bug report form. I have an idea though. If @Melvin-Abraham made a docs repo for this project, I can add a basic CODEDOC site with public issues addressed, then making a reply workflow here that responds to every issue made.
Then, if anyone opens an issue here, the workflow will be triggered, and it can reply to the issue. This is live on my blog, you can test that. I will make a feature request *after* this PR gets merged as a test but a real feature request is made. Then, after the docs repo is made, I can make the website like I said above, make a PR, and if it gets merged, the gh-pages branch will be made, and after activation of GitHub Pages, it will be successfully deployed.

**Comment by im-coder-lg** [comment]
Alright, that was pure gobbledygook. What I mean is that if @Melvin-Abraham wants to address public issues, it is best done on a website. I can take care of the website work.

How I can do it:

1. Make a basic CODEDOC website
2. Get workflows and checks for the website
3. Make a PR at the docs repository
4. Add all public issues
5. After it gets merged and the gh-pages branch made with the built files, @Melvin-Abraham can enable GitHub Pages.
6. I make the reply workflow with a comment markdown file.
7. After the merge, and whenever ***anybody*** opens an issue, it will reply with a hyperlink to the GitHub Pages-deployed site, and people can see the roadmap, public issues addressed, a demonstration, setup guide for beginners, and et cetera.

**Comment by im-coder-lg** [comment]
But, is that really necessary? 🤔

**Comment by im-coder-lg** [comment]
About the feature request, I think I will be requesting for the reply workflow. And since I know *how* and *what* the code is, I can ask if I can do it since I know the entire code. Then, the reply workflow can be merged that gets activated on every new issue, and can give a link to the website/public issue page.

**Comment by sumeshir26** [comment]
@im-coder-lg, can't we just add a note at the top of the issue form?

**Comment by im-coder-lg** [comment]
I thought about it and this can be final. Add a link to the FAQ as markdown, so that if it's a common issue, they can click that and get the solution.

**Review by im-coder-lg — COMMENTED** [review]
You can find more configuration [<kbd>here</kbd>](https://gh-community.github.io/issue-template-feedback/structured/) and do not forget, do them in Gitpod because Gitpod showed pinpoint accuracy in mistakes. That time when I had errors, I used Gitpod to find out my mistakes.

File: .github/ISSUE_TEMPLATE/bug_report.yml
Comment (im-coder-lg) [reply .github/ISSUE_TEMPLATE/bug_report.yml:13]: Wait how can I commit this here?

**Comment by Melvin-Abraham** [comment]
This looks great! ❤
I'll make some minor changes and merge it.

**Comment by im-coder-lg** [comment]
Oh yeah, make sure that everything works. Even a small indentation error spooled everything up. And yeah you don't need the issue actions workflow, I have a better idea(commenter that has a link to the FAQs so that the user can see that as a comment as well as mentioning you so that you get a direct mail/notification.)

**Comment by im-coder-lg** [comment]
I know the entire code to it.

**Comment by im-coder-lg** [comment]
Demo: https://github.com/im-coder-lg/coder-lg-blog-codedoc/issues/56

**Comment by im-coder-lg** [comment]
The code: https://github.com/im-coder-lg/coder-lg-blog-codedoc/blob/master/.github/workflows/issue-and-project.yml
All you need is the Create Issue workflow. Ignore the triage part, but if you _really_ _really_ want it, make a Bug Triage project in this repo, add the column ID to the workflow, and you are good to go.

**Comment by sumeshir26** [comment]
@im-coder-lg I merged your markdown

**Review by im-coder-lg — APPROVED** [review]
All good! :+1: :smile:

**Comment by im-coder-lg** [comment]
> Now the note is there, but users are sometimes prone to ignore that. Moreover, there is no use for the issue actions workflow, it automatically assigns the issue to @Melvin-Abraham and I advise to remove it or else, the workflow might remove the `bug` label thinking that it added it.

If you see the issue forms' front matter, you can see that labels are added and you are added as an assignee. If I open an issue after this is merged, say, a Bug Report, you don't need extra CI to assign and add labels. Highly suggested: Remove that, and if you want, I will make a customized workflow as a PR that will comment(see my past replies here: https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/pull/503#issuecomment-910345198 and here: https://github.com/Melvin-Abraham/Google-Assistant-Unofficial-Desktop-Client/pull/503#issuecomment-910347575.

If you allow permission, I will make the workflow and open the PR up.

**Comment by sumeshir26** [comment]
@im-coder-lg But the CI still adds an extra label 'Duplicate' if it sees specific keywords

**Comment by im-coder-lg** [comment]
Yes, but since the CI adds the bug label, it might get removed. Moreover, the assign part is problematic. But, if we use GitHub Script and modify the workflow to scan the issue for specific keywords, it can apply the `duplicate` label.

**Comment by im-coder-lg** [comment]
GitHub Script is almost like JavaScript, except all you need to use is GitHub things like issues, PRs, etc.

**Comment by im-coder-lg** [comment]
That is what I am using for my workflow. There has to be some issue scanning component in GitHub Script. By using that, and making a scanner for specific keywords, you can make sure that duplicate issues get the duplicate label.

**Review by im-coder-lg — APPROVED** [review]
Perfect.

**Comment by im-coder-lg** [comment]
I think I will test this _after_ it gets merged socthat I can check for errors. Have you tested the code in Gitpod?

**Comment by im-coder-lg** [comment]
Because then you can know if there are errors.

**Comment by im-coder-lg** [comment]
All good, no errors and configure the Issue Actions Workflow as it might cause errors. Add the `duplicate` part only and remove the rest as it might remove the bug or enhancement label and even remove assign.

File: .github/ISSUE_TEMPLATE/feature_request.yml
Comment (Melvin-Abraham) [inline .github/ISSUE_TEMPLATE/feature_request.yml:66]: Do we really need this? Opinions?

Comment (sumeshir26) [reply .github/ISSUE_TEMPLATE/feature_request.yml:66]: @Melvin-Abraham It is not really necessary, if you ask me. Since I got the basic code from [files](https://github.com/files-community/files) it is from there.

Comment (im-coder-lg) [reply .github/ISSUE_TEMPLATE/feature_request.yml:66]: Maybe you can remove them

Comment (im-coder-lg) [reply .github/ISSUE_TEMPLATE/feature_request.yml:66]: They are from the issue form templates that was in GitHub Docs.

Comment (Melvin-Abraham) [reply .github/ISSUE_TEMPLATE/feature_request.yml:66]: Cool. I'll remove it then...

**Comment by Melvin-Abraham** [comment]
_Merging this PR_

**Comment by im-coder-lg** [comment]
👍🏻

**Pull request merged**
